doc/source/whatsnew/v1.1.0.rst

```diff
@@ -811,6 +811,7 @@ Deprecations
 - :meth:`DatetimeIndex.week` and `DatetimeIndex.weekofyear` are deprecated and will be removed in a future version, use :meth:`DatetimeIndex.isocalendar().week` instead (:issue:`33595`)
 - :meth:`DatetimeArray.week` and `DatetimeArray.weekofyear` are deprecated and will be removed in a future version, use :meth:`DatetimeArray.isocalendar().week` instead (:issue:`33595`)
 - :meth:`DateOffset.__call__` is deprecated and will be removed in a future version, use ``offset + other`` instead (:issue:`34171`)
+- :meth:`~BusinessDay.apply_index` is deprecated and will be removed in a  future version. Use ``offset + other`` instead (:issue:`34580`)
```

Comment:
should probably deprecate for all DateOffset subclasses?

Reply:
It is deprecated for all the subclasses. We don't have DateOffset.apply_index in the API docs (since it's not always implemented?). With the `~` this will render as just `.apply_index()` so hopefully it'll be clear that it applies to all DateOffset subclasses.